src/modules/form/components/DynamicField.tsx

```diff
@@ -410,20 +411,51 @@ const DynamicField: React.FC<DynamicFieldProps> = ({
     case ItemType.Image:
       return (
         <InputContainer {...commonContainerProps}>
-          <Uploader
-            id={linkId}
-            image
-            onUpload={async (upload) => onChangeValue(upload.signedBlobId)}
-          />
+          <LabelWithContent label={item.text} helperText={item.helperText}>
+            {item.repeats ? (
+              <Uploader
+                multiple={true}
+                id={linkId}
+                files={value}
+                image
+                onChange={onChangeValue}
+                ariaLabel={item.text}
+              />
+            ) : (
+              <Uploader
+                multiple={false}
+                id={linkId}
+                file={value}
+                image
+                onChange={onChangeValue}
+                ariaLabel={item.text}
+              />
+            )}
```

Comment:
Collapse this ternary and just conditionally pass `multiple`?
```suggestion
            <Uploader
              multiple={item.repeats || false}
              id={linkId}
              files={value}
              image
              onChange={onChangeValue}
              ariaLabel={item.text}
            />
```

Reply:
Hm let's look at this together, I agree it would be better to get rid of the ternary but I think that it means reverting some of the changes I made related to [this thread](https://github.com/greenriver/hmis-frontend/pull/982#discussion_r1856890379). With the current typescript regime on the Uploader component, getting rid of the ternary leads to unexpected behavior with an existing file on a single-uploader.

Reply:
Updated to use generics similar to Autocomplete. Some casting is still needed internally to the component to get typescript happy, but I think the api is way better. https://github.com/greenriver/hmis-frontend/pull/982/commits/9777a416f318f3ba4f998221d6ff0f87e123e892

Reply:
Nice improvement, thanks!